### PR TITLE
Setup Property Files

### PR DESCRIPTION
**Pull request by daekoon**
### Server Side
- All servlets now read from the properties file to obtain api key.
- Created a new servlet, `ClientConfigServlet` which allows clients to retrieve config values.

### Client Side
- Changed map initialization from static (HTML) to dynamic.
- Client now makes a get request to `client-config` endpoint to retrieve the API key.

### Other changes
- Added two template property files, `server.properties.sample` and `client.properties.sample`
- Added setup instructions to README

### CODE REVIEW

File: src/main/resources/server.properties.sample [hunk 1]
```diff
@@ -0,0 +1,2 @@
+serverapikey=YOUR_API_KEY_HERE
+clientapikey=YOUR_API_KEY_HERE
```
Comment (lolfuljames) [inline src/main/resources/server.properties.sample:2]: Where will this be used?

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -4,6 +4,19 @@ let markers = [];
...
+fetch(configUrl).then((response => {
+  response.text().then(apikey => {
+    var script = document.createElement('script');
+    script.src = 'https://maps.googleapis.com/maps/api/js?key=' + apikey + '&libraries=places&callback=initMap';
+    script.defer = true;
+
+    document.head.appendChild(script);
+  })
+}))
+
```
Comment (lolfuljames) [inline src/main/webapp/script.js]: Will this slow-down the initial load-up speed of the website?

**Review by lolfuljames — COMMENTED** [review]
The server-side implementation is great! This will save alot of our time, thanks! However, I am afraid that we might be over-engineering the client-side implementation.

We should discuss this in the meeting and get Henry's opinion on this as well

File: src/main/resources/server.properties.sample (hunk 1 above)
Comment (daekoon) [reply src/main/resources/server.properties.sample:2]: This file exists as a base template for the properties.sample file.

File: src/main/webapp/script.js (hunk 2 above)
Comment (daekoon) [reply src/main/webapp/script.js]: Currently I'm seeing around 100ms delay to load the config on my desktop. Loading the frontend config takes about the same time as making 1 query - so its around 12% increase in delay.
Do you think its worth keeping it in to save the hassle of having to deal with api key in source code?

Comment (lolfuljames) [reply src/main/webapp/script.js:19]: Nice, but I think having the restrictions on the API key would be sufficient. I'm on the fence on this. @henryken care to share your thoughts on this?

**Comment by henryken** [comment]
The most important thing is to protect the API key, e.g. by referer.
Having it stored on GitHub then is not that catastrophic.

Having it stored separately from the repo is definitely better, though it still doesn't eliminate the exposure of the key to public.

An option that may improve performance:
* To load the API key at the head without defer. This allows it to load first while the body is rendering.
* Have a cache control with reasonable expiry, to cater for the case when the key changes.

**Comment by daekoon** [comment]
I've decided to not handle properties file loading for the client side. There is no clean way of achieving this using base javascript, and since the client API key is restricted there shouldn't be much problem committing the code into the repository.

Maybe in the future when we decide to use other frontend stack (like at least node js), we can consider re-looking into this again.

**Review by leonyeo — APPROVED** [review]
LGTM

**Pull request merged**